File: day10.py
# Naod Philemon
# 12/10/2022

def part1():
    file = open('day10-input.txt')
    X, cycle, sum, check = 1, 0, 0, 20

    for line in file:
        if cycle > 220: break
        line = line.strip()

        cycle += 1
        if cycle % check == 0: 
            sum += cycle * X
            check += 40

        if line != 'noop':
            line = line.split()
            cycle += 1
            if cycle % check == 0: 
                sum += cycle * X
                check += 40
            X += int(line[1])
            
    return sum

def draw(cycle, X):
    if (cycle%40-1) <= X+1 and (cycle%40-1) >= X-1: 
        print('#', end='')
    else: print('.', end='')
    if cycle % 40 == 0: print()

def part2():
    print('Part 2')
    file = open('day10-input.txt')
    X, cycle = 1, 0
    
    for line in file:
        line = line.strip()
        cycle += 1
        draw(cycle, X)

        if line != 'noop':
            line = line.split()
            cycle += 1
            draw(cycle, X)
            X += int(line[1])
            
print('Part 1', part1())
print(part2())

File: Day 10/day10.cpp
// Naod Philemon
// 12/10/2022

#include <fstream>
#include <iostream>
#include <string>
using namespace std;

int part1() {
  ifstream file("day10-input.txt");
  int X = 1, cycle = 0, sum = 0, check = 20;

  string line;
  while (getline(file, line)) {
    if (cycle > 220)
      break;

    cycle++;
    if (cycle % check == 0) {
      sum += cycle * X;
      check += 40;
    }

    if (line != "noop") {
      cycle++;
      if (cycle % check == 0) {
        sum += cycle * X;
        check += 40;
      }
      X += stoi(line.substr(line.find(" ")));
    }
  }

  return sum;
}

void draw(int cycle, int X) {
  if ((cycle % 40 - 1) <= X + 1 && (cycle % 40 - 1) >= X - 1) {
    cout << "#";
  } else {
    cout << ".";
  }
  if (cycle % 40 == 0)
    cout << endl;
}

void part2() {
  ifstream file("day10-input.txt");
  int X = 1, cycle = 0;

  string line;
  while (getline(file, line)) {
    cycle++;
    draw(cycle, X);

    if (line != "noop") {
      cycle++;
      draw(cycle, X);
      X += stoi(line.substr(line.find(" ")));
    }
  }
}

int main() {
  cout << "Part 1 " << part1() << endl;
  cout << "Part 2 " << endl;
  part2();

  return 0;
}
